hudi-sync/hudi-sync-common/src/main/java/org/apache/hudi/sync/common/util/SyncUtilHelpers.java

```diff
@@ -104,4 +107,22 @@ static HoodieSyncTool instantiateMetaSyncTool(String syncToolClassName,
           + ": no valid constructor found.");
     }
   }
+
+  public static HoodieException getExceptionFromList(Collection<HoodieException> exceptions) {
+    if (exceptions.size() == 1) {
+      return exceptions.stream().findFirst().get();
+    }
+    StringBuilder sb = new StringBuilder();
+    sb.append("Multiple exceptions during meta sync:\n********\n");
+    exceptions.forEach(e -> {
+      StringWriter sw = new StringWriter();
+      PrintWriter pw = new PrintWriter(sw);
+      e.printStackTrace(pw);
```

Comment:
Let's avoid printStacktrace and just append exception.getMessage() to string builder.